internal/pkg/consensus/validation.go

```diff
@@ -127,32 +114,59 @@ func canCoverGasLimit(msg *types.UnsignedMessage, actor *actor.Actor) bool {
 	return actor.Balance.GreaterThanEqual(expense)
 }
 
-// IngestionValidatorAPI allows the validator to access latest state
-type ingestionValidatorAPI interface {
-	Head() block.TipSetKey
-	GetActorAt(ctx context.Context, tipKey block.TipSetKey, addr address.Address) (*actor.Actor, error)
-	AccountStateView(baseKey block.TipSetKey) (state.AccountStateView, error)
+// MessageSyntaxValidator checks basic conditions independent of current state
+type MessageSyntaxValidator struct{}
+
+func NewMessageSyntaxValidator() *MessageSyntaxValidator {
+	return &MessageSyntaxValidator{}
+}
+
+func (v *MessageSyntaxValidator) Validate(ctx context.Context, smsg *types.SignedMessage) error {
+	// check non-state dependent invariants
+	msg := smsg.Message
+	if msg.GasLimit > types.BlockGasLimit {
+		log.Debugf("Message: %s gas limit from actor: %s above block limit: %s", msg, msg.From, types.BlockGasLimit)
+		errGasAboveBlockLimitCt.Inc(ctx, 1)
+		return errGasAboveBlockLimit
+	}
+
+	if msg.Value.LessThan(specsbig.Zero()) {
+		log.Debugf("Cannot transfer negative value: %s from actor: %s", msg.Value, msg.From)
+		errNegativeValueCt.Inc(ctx, 1)
+		return errNegativeValue
+	}
+
+	if msg.From == msg.To {
+		return errSelfSend
+	}
```

Comment:
This is now allowed by the spec. I assume that your goal was not to fully implement message syntax validation since there are other divergences too, and that's fine. But please ensure an issue remains to complete this.

Reply:
Ah didn't realize, I'll remove